modules/ppcp-button/resources/js/modules/Helper/WaitForElement.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Waits for a DOM element using setTimeout polling
+ * @param {string} selector - CSS selector for the element
+ * @param {number} timeout  - Maximum time to wait in milliseconds (default: 3000)
+ * @param {number} interval - Polling interval in milliseconds (default: 100)
+ * @return {Promise<Element>} - Resolves with the element or rejects if timeout
+ */
+export function waitForElement( selector, timeout = 3000, interval = 100 ) {
```

Comment:
Feels more like an util to me rather than a helper, but not a strong opinion here.